src/OSPSuite.Infrastructure.Import/Services/IndividualValuesCacheImporter.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using LumenWorks.Framework.IO.Csv;
+using OSPSuite.Assets;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Populations;
+using OSPSuite.Core.Extensions;
+using OSPSuite.Core.Services;
+using OSPSuite.Infrastructure.Import.Extensions;
+using OSPSuite.Utility.Collections;
+using OSPSuite.Utility.Extensions;
+
+namespace OSPSuite.Infrastructure.Import.Services
+{
+   public interface IIndividualValuesCacheImporter
+   {
+      /// <summary>
+      ///    Imports all parameters defined in the <paramref name="populationFileFullPath" />. If
+      ///    <paramref name="allParameters" /> is defined,
+      ///    the imported parameter path will be validated (unit removed, check for existence). Otherwise, parameters will be
+      ///    imported as is
+      /// </summary>
+      /// <param name="populationFileFullPath">Full path of CSV File containing the population to import</param>
+      /// <param name="logger">Logger used to notify user of potential problem with the file</param>
+      /// <param name="allParameters">Optional cache of parameters used to validate the imported parameters</param>
+      /// <returns></returns>
+      IndividualValuesCache ImportFrom(string populationFileFullPath, IImportLogger logger, PathCache<IParameter> allParameters = null);
+   }
+
+   public class IndividualValuesCacheImporter : IIndividualValuesCacheImporter
+   {
+      private static readonly char[] ALLOWED_DELIMITERS = {',', ';', '\t'};
+
+      public IndividualValuesCache ImportFrom(string populationFileFullPath, IImportLogger logger, PathCache<IParameter> allParameters = null)
+      {
+         try
+         {
+            foreach (var delimiter in ALLOWED_DELIMITERS)
+            {
+               var individualValuesCache = individualValuesCacheFrom(populationFileFullPath, delimiter);
+               //we found at least one individual, this is a valid file for the delimiter and we can exit
+               if (individualValuesCache?.Count > 0)
+                  return withPathsContainingUnitsUpdated(individualValuesCache, allParameters, logger);
+            }
+
+            //no match. Log 
+            logger.AddError(Warning.PopulationFileFormatIsNotSupported);
+            return new IndividualValuesCache();
+         }
+         catch (Exception e)
+         {
+            logger.AddError(e.FullMessage());
+            return new IndividualValuesCache();
+         }
+      }
+
+      private IndividualValuesCache withPathsContainingUnitsUpdated(IndividualValuesCache individualValuesCache, PathCache<IParameter> allParameters, IImportLogger logger)
+      {
+         // No parameters to check from, return as IS
+         if (allParameters == null)
```

Comment:
This is the case from R. In PKSim, the list will be given and as  such, the parameters will be striped from units